### PR TITLE
Mobile wallet signature confirmation modal

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6719

## Description of Changes
- Added a signature verification modal when logging-in with wallet connect on mobile devices

image

## "How We Fixed It"
N/A

## Test Plan
- Enable `FLAG_NEW_SIGN_IN_MODAL` in `.env`
- On your mobile device, visit the site, logout and go to any Ethereum based community or any non-community page 
- Attempt to login with wallet connect
- Verify you see the signature verification modal on mobile
- Verify that clicking on the `Sign with wallet` button opens your selected wallet for signature confirmation
- Verify that confirming the signature from wallet redirects back to app, logs you in, and closes AuthModal

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

**Review by masvelio — COMMENTED** [review]
https://github.com/hicommonwealth/commonwealth/assets/14819225/cff846e3-9950-41a4-9245-7b89f7b48089

Tried couple times but could not see the modal even once. However was able to sign in, because I tapped `sign` with metamask UI. So a bit confused here 🤔

File: packages/commonwealth/client/scripts/hooks/useWallets.tsx [hunk 1]
```diff
@@ -745,6 +745,24 @@ const useWallets = (walletProps: IuseWalletProps) => {
...
+  }, [isMobile, activeStep]);
+  const onVerifyMobileWalletSignature = async () => {
+    await onAccountVerified(
+      signerAccount,
+      isNewlyCreated,
+      isLinkingWallet,
+      selectedWallet,
+    );
+    setIsMobileWalletVerificationStep(false);
+    await walletProps?.onModalClose?.();
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/client/scripts/hooks/useWallets.tsx:763]: nice spot, in some instances like [this](https://github.com/hicommonwealth/commonwealth/blob/e1340bf9a419a420769088634a9dafec41c1634a/packages/commonwealth/client/scripts/views/modals/AuthModal/BaseModal/BaseModal.tsx#L40-L58), `onModalClose` is a promise, while not in other's. To be safe i added a new commit to call all `walletProps?.onModalClose?.()` inside `useWallets` with `await`

**Comment by mzparacha** [comment]
> I tried debugging this multiple times with different wallets, but I always get the confirmation modal, not sure what is going on. I will ask others to test if they get the same.

Others were also not getting the modal to show up, finding a fix

**Comment by mzparacha** [comment]
Updated, should be good to test now.

**Comment by CowMuon** [comment]
@egetekiner to review and test please.

**Comment by masvelio** [comment]
https://github.com/hicommonwealth/commonwealth/assets/14819225/bd8ba692-03d4-43a5-9a80-48cd32207bd5

The new modal appears but after successful login, main login modal is still visible for me, even though I can see that I am signed in already in the background.

**Comment by mzparacha** [comment]
Thanks for the review, i added a fix for it earlier but looks like it doesn't work in all cases. Trying for a better solution.

**Comment by mzparacha** [comment]
Ok so should be fixed now. The issue was that we were resetting wallet connect after login, and sometimes due to delays b/w mobile wallet and app connection it was throwing connection error after login which caused the `onClose` to sometimes fail. I added a `.catch(() => null)` to it and it should work correctly now. 

As a sidenote: The resetting of wallet connect after login is not necessarily needed, but its good in cases when user authentication attempts repeatedly fail due to local wallet connect cookies, which we try to reset after login. 

cc: @masvelio

**Review by masvelio — APPROVED** [review]
works fine for me now!

**Review by burtonator — APPROVED** [review]

**Pull request merged**